baskup

```diff
@@ -19,41 +19,41 @@ while getopts ":a" opt; do
   esac
 done
 
-function select_rows () {
+function sql () {
   sqlite3 ~/Library/Messages/chat.db "$1"
 }
 
-for line in $(select_rows "select distinct guid from chat;" ); do
```

Comment:
This is replaced by the `read` because this `for` relies on "word splitting" while the sqlite3 output is line-based.  It's a disagreement between output and parsing that can lead to bugs.  Using `read`, the input is parsed in a line-based manner.  `IFS` is set empty to prevent `read`'s default behaviour of trimming whitespace from the input.  It's essentially an idiom, not necessarily tweaked specifically from a requirement of this specific sqlite3's output.